draft-ietf-jsonpath-base.md

```diff
@@ -1298,6 +1294,7 @@ The `dot-wildcard-shorthand` is shorthand for `[*]`.
 
 A `dot-member-name-shorthand` of the form `.<member-name>` is shorthand for `['<member-name>']`, but
 can only be used with member names that are composed of certain characters.
+Thus, for example, `$.foo.bar` is shorthand for `$['foo']['bar']` (and not for `$['foo.bar']`).
```

Comment:
Would still like to see a caution for developers against using `"$" + join(segments, ".")`

Reply:
This caution should then also apply to "$" + segments.map { "['" + _1 + "']" }.join
What would be a good place for this?
(Security considerations comes to mind.)